wurst/util/UnitUtils.wurst

```diff
@@ -0,0 +1,120 @@
+package UnitUtils
+import BuffObjEditing
+import ClosureTimers
+
+
+
+/*
+
+This file contains various utility functions for Unit, which require enhanced functionality like closures and cannot be defined in Unit.wurst
+
+*/
+
+@configurable public constant real FIELD_OF_VIEW_FRONT  = 115.00  // Real FoV is: 180 - 2x (FIELD_OF_VIEW-90)	Using a value of 115 generates a FoV of 130   (natural FoV is about 135 degrees)
+@configurable public constant real FIELD_OF_VIEW_BEHIND = 115.00 
+
+/** Apply Buff to this unit for duration.*/
+public function unit.applyBuffTimed(buffTuple buffApplied, real duration)
+	this.addAbility(buffApplied.abilId)
+	this.setAbilityLevel(buffApplied.abilId, 1)
+	doAfter(duration) ->
+		if this != null
+			this.setAbilityLevel(buffApplied.abilId, 0)
+			this.removeAbility(buffApplied.abilId)
+			
+/** Increase Armor of a unit for a duration */
+public function unit.addArmorTimed(real armorAdd, real duration)
```

Comment:
This and all other Timed things sounds like a very specific mechanic implementation.
e: aside from applyBuffTimed thing, that thing is good.

Reply:
you say this is to specific but the "applyBuffTimed" is good it is the same more or less, in both are special mechanics, so either add both with PR or non i would say

Reply:
As Krypton mentions, I'd see this easily connected with any ability.
Cast ability on ally, which applyBuffTimed(x) and also addArmorTimed(Y, x) on given unit. Ofc this is useful only for coded spells, but atleast player does not have to define such function himself.